src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,303 @@
+type SettableTableProps = Pick<TableProps, "size" | "stickyHeader">;
+
+interface Column {
+    alias?: string;
+    align?: TableCellProps["align"];
+    format?: (value: any) => any;
+    parse?: (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: Row,
+        column: Column
+    ) => any;
+    includeTotal?: boolean;
+    editable: boolean;
+    type: string;
+    name: string;
+}
+
+interface EditableTableElementProps
+    extends FormElementProps<Row[]>,
+        SettableBoxProps,
+        SettableTableProps {
+    cols: Column[];
+    rows: Row[];
+    editTooltip?: string;
+    saveTooltip?: string;
+    includeTotals?: boolean;
+    footerLabel?: string;
+    footerStyle?: CSSProperties;
+    saveIcon?: string;
+    editIcon?: string;
+    onMouseEnter?: (row: Row) => void;
+    onMouseLeave?: (row: Row) => void;
+    onChange?: (data: Row[], row: Row) => void;
+}
+
+const IconButtonStyleOverrides: IconButtonProps["sx"] = {
+    // Hack required to override .gcx-forms.defaults
+    borderRadius: "50% !important",
+    // Hack required to match disabled list item content
+    "&.Mui-disabled": {
+        opacity: 0.38,
+    },
+};
+
+const calculateTotals = (cols: Column[], rows: Row[]): Total => {
+    const totalCols = cols.filter((x) => x.includeTotal);
+    const total: Total = {};
+    for (const col of totalCols) {
+        const vals = rows.map((x) => x[col.name]);
+        const numericVals = vals.filter((x) => typeof x == "number");
+        const sum =
+            numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+        total[col.name] = sum;
+    }
+    return total;
+};
+
+/**
+ * A table form element.
+ * @displayName Editable Table
+ * @description A table form element that allows editing.
+ * @param props The props that will be provided by the Workflow runtime.
+ */
+function EditableTableElement(props: EditableTableElementProps): React.ReactElement | null {
+    const {
+        rows,
+        cols,
+        maxHeight,
+        maxWidth,
+        onMouseEnter,
+        onMouseLeave,
+        onChange,
+        raiseEvent,
+        setValue,
+        size,
+        editTooltip,
+        saveTooltip,
+        stickyHeader,
+        footerLabel,
+        footerStyle,
+        saveIcon,
+        editIcon,
+    } = props;
+    const [data, setData] = useState(rows);
+    setValue(data);
+    useEffect(() => {
+        setData(rows);
+    }, [rows]);
+
+    const includeTotals = cols.some((x) => x.includeTotal);
+    const totalData: Total = calculateTotals(cols, rows);
+    const footerLbl = footerLabel ?? "Total";
+    const editIconSrc: string = editIcon ?? "edit";
+    const saveIconSrc: string = saveIcon ?? "save";
+    const editTitle = editTooltip ?? "Edit";
+    const saveTitle = saveTooltip ?? "Save";
+
+    const handleEditToggleClick = (event: React.MouseEvent, row: Row) => {
+        if (row.isEditing) {
```

Comment:
`row` is supposed to be a piece of data from the workflow. We can't use it to store the `isEditing` state of our component.

Reply:
Eliminating the toggle should make this go away.